src/content/learn/your-first-component.md

```diff
@@ -152,95 +152,93 @@ img { margin: 0 10px 10px 0; height: 90px; }
+### Ugnježdavanje i organizacija komponenata {/*nesting-and-organizing-components*/}
 
-Components are regular JavaScript functions, so you can keep multiple components in the same file. This is convenient when components are relatively small or tightly related to each other. If this file gets crowded, you can always move `Profile` to a separate file. You will learn how to do this shortly on the [page about imports.](/learn/importing-and-exporting-components)
+Komponente su obične JavaScript funkcije, tako da možete imati više komponenata u jednom fajlu. Ovo je zgodno kada su komponente relativno male ili su usko povezane međusobno. Ako fajl postane prenatrpan, uvek možete pomeriti `Profile` u poseban fajl. Naučićete kako se to radi ubrzo na [stranici o import-ovanju](/learn/importing-and-exporting-components).
 
-Because the `Profile` components are rendered inside `Gallery`—even several times!—we can say that `Gallery` is a **parent component,** rendering each `Profile` as a "child". This is part of the magic of React: you can define a component once, and then use it in as many places and as many times as you like.
+Pošto su `Profile` komponente renderovane unutar `Gallery`-a (i to više puta), možemo reći da je `Gallery` **roditeljska komponenta** koja renderuje svaki `Profile` kao "dete". Ovo je deo React-ove magije: komponentu definišete jednom, a onda je koristite koliko god i gde god želite.
 
 <Pitfall>
 
-Components can render other components, but **you must never nest their definitions:**
+Komponente mogu renderovati druge komponente, ali **nikad ne smete ugnježdavati njihove definicije**:
 
 ```js {2-5}
 export default function Gallery() {
-  // 🔴 Never define a component inside another component!
+  // 🔴 Nikada nemojte definisati komponentu unutar druge komponente!
   function Profile() {
     // ...
   }
   // ...
 }
 ```
 
-The snippet above is [very slow and causes bugs.](/learn/preserving-and-resetting-state#different-components-at-the-same-position-reset-state) Instead, define every component at the top level:
+Snippet iznad je [veoma spor i prouzrokuje bug-ove](/learn/preserving-and-resetting-state#different-components-at-the-same-position-reset-state). Umesto toga, definišite svaku komponentu na najvišem nivou unutar fajla:
 
 ```js {5-8}
 export default function Gallery() {
   // ...
 }
 
-// ✅ Declare components at the top level
+// ✅ Definišite komponente na najvišem nivou unutar fajla
 function Profile() {
   // ...
 }
 ```
 
-When a child component needs some data from a parent, [pass it by props](/learn/passing-props-to-a-component) instead of nesting definitions.
+Kada detetu (child) komponenti trebaju podaci od roditelja (parent), [prosledite ih preko props-a](/learn/passing-props-to-a-component) umesto da ugnježdavate definicije.
 
 </Pitfall>
 
 <DeepDive>
 
-#### Components all the way down {/*components-all-the-way-down*/}
+#### Komponente nemaju granice {/*components-all-the-way-down*/}
```

Comment:
This doesn't sound as the best translation, but I don't know any better way for `all the way down` phrase.

Reply:
leet it be as it is for now :D